File: src/main/java/com/chengte99/Ticket.java
package com.chengte99;

public class Ticket {
    // Model, JavaBean, POJO
    // Command + N 可以自動產生建構子，getter, setter
    int source;
    int destination;
    int price;

    public Ticket(int source, int destination, int price) {
        this.source = source;
        this.destination = destination;
        this.price = price;
    }

    public Ticket() {
    }

    public int getSource() {
        return source;
    }

    public void setSource(int source) {
        this.source = source;
    }

    public int getDestination() {
        return destination;
    }

    public void setDestination(int destination) {
        this.destination = destination;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}

File: src/main/kotlin/com/chengte99/Ticket.kt
package com.chengte99.kotlin

import java.util.*

data class Ticket(var source: Int, var destination: Int, var price: Int)
// Model

fun main() {
    val ticket = Ticket(20, 51, 330)
    println(ticket.price)

    val s = "abcdefg"
    println(s.validate())

    val range = 1..10 // IntRange為kotlin 範圍類別
    println(range.ramdom())
}

// extension 擴充原有類別的方法
fun String.validate(): Boolean {
    return this.length >= 6
    // this代表物件
}

fun IntRange.ramdom(): Int {
    // endInclusive跟start 為kotlin IntRange的屬性
    val r = Random().nextInt(endInclusive - start) + start
    return r
}

